aws/middleware/metadata.go

```diff
@@ -50,67 +50,103 @@ type (
 )
 
 // GetServiceID retrieves the service id from the context.
+//
+// Scoped to stack values. Use github.com/aws/smithy-go/middleware#ClearStackValues
+// to clear all stack values.
 func GetServiceID(ctx context.Context) (v string) {
-	v, _ = ctx.Value(serviceIDKey{}).(string)
+	v, _ = middleware.GetStackValue(ctx, serviceIDKey{}).(string)
```

Comment:
nit: should we be using smithymiddle as alias - just coz we are already in middleware pkg

Reply:
It might be good for SDK to always alias smithymiddle to be clear and readable.

Reply:
i think the `middleware` name has generally been the smithy middleware throughout the sdk the `aws/middleware` has been the one that has been aliased.

Reply:
I often see `smithymiddle` being used

Reply:
Looks like both are used pretty often in the SDK. `smithymiddle` was introduced in retry and endpoint creds